app/concerns/writable.rb

```diff
@@ -16,7 +16,7 @@ def has_icons?
       return user.avatar_id? unless character
       return true if character.default_icon
       return false unless character.galleries.present?
-      return character.galleries.map(&:icons).flatten.present?
+      return character.galleries.includes(:icons).map(&:icons).flatten.present?
```

Comment:
Fixing an N+1 query – might be possible to do better but I haven't touched this code before.